src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.HandleMetadata.cs

```diff
@@ -336,6 +336,11 @@ public static partial class JsonSerializer
 
             if (state.Current.ObjectState == StackFrameObjectState.ReadValuesStartArray)
             {
+                // Temporary workaround for the state machine accidentally
```

Comment:
Temporary until what?

Reply:
> Temporarily address a corner case in the metadata reader async state machine implementation. To be addressed properly in an upcoming refactoring of the metadata reader components.

I'm rewriting this particular code in an upcoming PR, but needed to get the new test coverage green beforehand.